File: analysis/R/02_descriptive_stats.R
# ==============================================================================
# 02_descriptive_stats.R
# Purpose: Compute and display summary statistics for the crossover dataset,
#          broken down by group, period, condition, and sequence.
# Input:   df_clean.rds from 01_data_import_clean.R
# Output:  Summary tables (CSV) and basic plots (PNG) in output/
# ==============================================================================

cat("=== 02_descriptive_stats.R: Computing descriptive statistics ===\n")

# Load cleaned data
df_clean <- readRDS(file.path(OUTPUT_DIR, "df_clean.rds"))

# --------------------------------------------------------------------------
# 1. Overall summary statistics
# --------------------------------------------------------------------------

overall_summary <- df_clean %>%
  summarise(
    n          = n(),
    mean_score = mean(score, na.rm = TRUE),
    sd_score   = sd(score, na.rm = TRUE),
    median_score = median(score, na.rm = TRUE),
    min_score  = min(score, na.rm = TRUE),
    max_score  = max(score, na.rm = TRUE),
    mean_rubric = mean(rubric_score, na.rm = TRUE),
    sd_rubric   = sd(rubric_score, na.rm = TRUE),
    mean_time  = mean(time_spent, na.rm = TRUE),
    sd_time    = sd(time_spent, na.rm = TRUE)
  )

cat("  Overall summary:\n")
print(overall_summary)

# --------------------------------------------------------------------------
# 2. Summary by condition (AI vs noAI)
# --------------------------------------------------------------------------

by_condition <- df_clean %>%
  group_by(condition) %>%
  summarise(
    n          = n(),
    mean_score = round(mean(score, na.rm = TRUE), 2),
    sd_score   = round(sd(score, na.rm = TRUE), 2),
    median_score = round(median(score, na.rm = TRUE), 2),
    mean_rubric = round(mean(rubric_score, na.rm = TRUE), 2),
    sd_rubric   = round(sd(rubric_score, na.rm = TRUE), 2),
    mean_time  = round(mean(time_spent, na.rm = TRUE), 2),
    sd_time    = round(sd(time_spent, na.rm = TRUE), 2),
    .groups    = "drop"
  )

cat("\n  Summary by condition:\n")
print(by_condition)

write_csv(by_condition, file.path(OUTPUT_DIR, "tables", "summary_by_condition.csv"))

# --------------------------------------------------------------------------
# 3. Summary by condition and period (the 2x2 cell means)
# --------------------------------------------------------------------------

by_condition_period <- df_clean %>%
  group_by(condition, period) %>%
  summarise(
    n          = n(),
    mean_score = round(mean(score, na.rm = TRUE), 2),
    sd_score   = round(sd(score, na.rm = TRUE), 2),
    mean_rubric = round(mean(rubric_score, na.rm = TRUE), 2),
    sd_rubric   = round(sd(rubric_score, na.rm = TRUE), 2),
    .groups    = "drop"
  )

cat("\n  Summary by condition x period:\n")
print(by_condition_period)

write_csv(by_condition_period,
          file.path(OUTPUT_DIR, "tables", "summary_condition_period.csv"))

# --------------------------------------------------------------------------
# 4. Summary by sequence (AB vs BA)
# --------------------------------------------------------------------------

by_sequence <- df_clean %>%
  group_by(sequence, period, condition) %>%
  summarise(
    n          = n(),
    mean_score = round(mean(score, na.rm = TRUE), 2),
    sd_score   = round(sd(score, na.rm = TRUE), 2),
    .groups    = "drop"
  )

cat("\n  Summary by sequence x period x condition:\n")
print(by_sequence)

write_csv(by_sequence, file.path(OUTPUT_DIR, "tables", "summary_by_sequence.csv"))

# --------------------------------------------------------------------------
# 5. Basic plots
# --------------------------------------------------------------------------

# Boxplot: Score by condition
p1 <- ggplot(df_clean, aes(x = condition, y = score, fill = condition)) +
  geom_boxplot(alpha = 0.7, outlier.shape = 21) +
  geom_jitter(width = 0.15, alpha = 0.3, size = 1) +
  scale_fill_brewer(palette = "Set2") +
  labs(
    title = "Score Distribution by Condition",
    x     = "Condition",
    y     = "Score (0-100)"
  ) +
  theme(legend.position = "none")

ggsave(file.path(OUTPUT_DIR, "figures", "boxplot_score_by_condition.png"),
       p1, width = 6, height = 5, dpi = 300)

# Boxplot: Score by condition and period
p2 <- ggplot(df_clean, aes(x = period, y = score, fill = condition)) +
  geom_boxplot(alpha = 0.7, position = position_dodge(0.8)) +
  scale_fill_brewer(palette = "Set2") +
  labs(
    title = "Score by Period and Condition",
    x     = "Period",
    y     = "Score (0-100)",
    fill  = "Condition"
  )

ggsave(file.path(OUTPUT_DIR, "figures", "boxplot_score_period_condition.png"),
       p2, width = 7, height = 5, dpi = 300)

# Histogram of score differences (AI - noAI)
df_wide <- readRDS(file.path(OUTPUT_DIR, "df_wide.rds"))

p3 <- ggplot(df_wide, aes(x = score_diff)) +
  geom_histogram(binwidth = 5, fill = "steelblue", color = "white", alpha = 0.8) +
  geom_vline(xintercept = 0, linetype = "dashed", color = "red", linewidth = 0.8) +
  geom_vline(xintercept = mean(df_wide$score_diff, na.rm = TRUE),
             linetype = "solid", color = "darkblue", linewidth = 0.8) +
  labs(
    title    = "Distribution of Within-Subject Score Differences (AI - noAI)",
    subtitle = "Blue line = mean difference; red dashed = zero",
    x        = "Score difference (AI - noAI)",
    y        = "Count"
  )

ggsave(file.path(OUTPUT_DIR, "figures", "histogram_score_diff.png"),
       p3, width = 7, height = 5, dpi = 300)

# Time spent by condition
p4 <- ggplot(df_clean, aes(x = condition, y = time_spent, fill = condition)) +
  geom_boxplot(alpha = 0.7) +
  scale_fill_brewer(palette = "Pastel1") +
  labs(
    title = "Time Spent by Condition",
    x     = "Condition",
    y     = "Time (minutes)"
  ) +
  theme(legend.position = "none")

ggsave(file.path(OUTPUT_DIR, "figures", "boxplot_time_by_condition.png"),
       p4, width = 6, height = 5, dpi = 300)

cat("  Saved tables and figures to output/\n")
cat("=== 02_descriptive_stats.R: Complete ===\n\n")

File: analysis/python/02_descriptive_stats.py
"""
02_descriptive_stats.py
Purpose: Compute and display summary statistics for the crossover dataset,
         broken down by group, period, condition, and sequence.
Input:   df_clean from 01_data_import_clean.py
Output:  Summary tables (CSV) and basic plots (PNG) in output/
"""

# ---------------------------------------------------------------------------
# Imports and setup
# ---------------------------------------------------------------------------
exec(open(str(__import__("pathlib").Path(__file__).resolve().parent / "00_setup.py")).read())

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

print("=== 02_descriptive_stats.py: Computing descriptive statistics ===")

# ---------------------------------------------------------------------------
# Load cleaned data
# ---------------------------------------------------------------------------

df_clean = pd.read_parquet(OUTPUT_DIR / "df_clean.parquet")
df_wide = pd.read_parquet(OUTPUT_DIR / "df_wide.parquet")

# Restore categoricals
df_clean["condition"] = pd.Categorical(df_clean["condition"], categories=["noAI", "AI"])
df_clean["period"] = pd.Categorical(df_clean["period"], categories=["Period 1", "Period 2"])
df_clean["sequence"] = pd.Categorical(df_clean["sequence"], categories=["AB", "BA"])

# ---------------------------------------------------------------------------
# 1. Overall summary statistics
# ---------------------------------------------------------------------------

overall = df_clean[["score", "rubric_score", "time_spent"]].describe().round(2)
print("  Overall summary:")
print(overall.to_string())

# ---------------------------------------------------------------------------
# 2. Summary by condition (AI vs noAI)
# ---------------------------------------------------------------------------

by_condition = (
    df_clean.groupby("condition", observed=True)
    .agg(
        n=("score", "size"),
        mean_score=("score", "mean"),
        sd_score=("score", "std"),
        median_score=("score", "median"),
        mean_rubric=("rubric_score", "mean"),
        sd_rubric=("rubric_score", "std"),
        mean_time=("time_spent", "mean"),
        sd_time=("time_spent", "std"),
    )
    .round(2)
    .reset_index()
)

print("\n  Summary by condition:")
print(by_condition.to_string(index=False))
by_condition.to_csv(TABLES_DIR / "summary_by_condition.csv", index=False)

# ---------------------------------------------------------------------------
# 3. Summary by condition and period (the 2x2 cell means)
# ---------------------------------------------------------------------------

by_cond_period = (
    df_clean.groupby(["condition", "period"], observed=True)
    .agg(
        n=("score", "size"),
        mean_score=("score", "mean"),
        sd_score=("score", "std"),
        mean_rubric=("rubric_score", "mean"),
        sd_rubric=("rubric_score", "std"),
    )
    .round(2)
    .reset_index()
)

print("\n  Summary by condition x period:")
print(by_cond_period.to_string(index=False))
by_cond_period.to_csv(TABLES_DIR / "summary_condition_period.csv", index=False)

# ---------------------------------------------------------------------------
# 4. Summary by sequence
# ---------------------------------------------------------------------------

by_sequence = (
    df_clean.groupby(["sequence", "period", "condition"], observed=True)
    .agg(
        n=("score", "size"),
        mean_score=("score", "mean"),
        sd_score=("score", "std"),
    )
    .round(2)
    .reset_index()
)

print("\n  Summary by sequence x period x condition:")
print(by_sequence.to_string(index=False))
by_sequence.to_csv(TABLES_DIR / "summary_by_sequence.csv", index=False)

# ---------------------------------------------------------------------------
# 5. Basic plots
# ---------------------------------------------------------------------------

# -- Boxplot: Score by condition --
fig, ax = plt.subplots(figsize=(6, 5))
sns.boxplot(data=df_clean, x="condition", y="score", palette=PALETTE, ax=ax,
            flierprops=dict(marker="o", markerfacecolor="grey", alpha=0.5))
sns.stripplot(data=df_clean, x="condition", y="score", color="black",
              alpha=0.2, size=3, jitter=True, ax=ax)
ax.set_title("Score Distribution by Condition")
ax.set_xlabel("Condition")
ax.set_ylabel("Score (0-100)")
fig.tight_layout()
fig.savefig(FIGURES_DIR / "boxplot_score_by_condition.png")
plt.close(fig)

# -- Boxplot: Score by condition and period --
fig, ax = plt.subplots(figsize=(7, 5))
sns.boxplot(data=df_clean, x="period", y="score", hue="condition",
            palette=PALETTE, ax=ax)
ax.set_title("Score by Period and Condition")
ax.set_xlabel("Period")
ax.set_ylabel("Score (0-100)")
ax.legend(title="Condition", loc="upper left")
fig.tight_layout()
fig.savefig(FIGURES_DIR / "boxplot_score_period_condition.png")
plt.close(fig)

# -- Histogram of score differences (AI - noAI) --
mean_diff = df_wide["score_diff"].mean()

fig, ax = plt.subplots(figsize=(7, 5))
ax.hist(df_wide["score_diff"].dropna(), bins=15, color="steelblue",
        edgecolor="white", alpha=0.8)
ax.axvline(0, linestyle="--", color="red", linewidth=0.8, label="Zero")
ax.axvline(mean_diff, linestyle="-", color="darkblue", linewidth=0.8,
           label=f"Mean = {mean_diff:.1f}")
ax.set_title("Distribution of Within-Subject Score Differences (AI - noAI)")
ax.set_xlabel("Score difference (AI - noAI)")
ax.set_ylabel("Count")
ax.legend()
fig.tight_layout()
fig.savefig(FIGURES_DIR / "histogram_score_diff.png")
plt.close(fig)

# -- Time spent by condition --
fig, ax = plt.subplots(figsize=(6, 5))
sns.boxplot(data=df_clean, x="condition", y="time_spent",
            palette={"noAI": "#fbb4ae", "AI": "#b3cde3"}, ax=ax)
ax.set_title("Time Spent by Condition")
ax.set_xlabel("Condition")
ax.set_ylabel("Time (minutes)")
fig.tight_layout()
fig.savefig(FIGURES_DIR / "boxplot_time_by_condition.png")
plt.close(fig)

print("  Saved tables and figures to output/")
print("=== 02_descriptive_stats.py: Complete ===\n")
